singer_sdk/testing.py

```diff
@@ -40,7 +40,52 @@ def _test_stream_connections() -> None:
         tap1: Tap = tap_class(config=config, parse_env_config=True)
         tap1.run_connection_test()
 
-    return [_test_cli_prints, _test_discovery, _test_stream_connections]
+    def _test_pkeys_in_schema() -> None:
+        """Verify that primary keys are actually in the stream's schema."""
+        tap = tap_class(config=config, parse_env_config=True)
+        for name, stream in tap.streams.items():
+            pkeys = stream.primary_keys or []
+            schema_props = set(stream.schema["properties"].keys())
+            for pkey in pkeys:
+                error_message = (
+                    f"Coding error in stream '{name}': " +
+                    f"primary_key '{pkey}' is missing in schema"
```

Comment:
```suggestion
                    f"Coding error in stream '{name}': "
                    + f"primary_key '{pkey}' is missing in schema"
```

Reply:
@laurentS I think this is the only thing missing for `pre-commit` to pass 🙂

Reply:
@edgarrmondragon sorry I've been incredibly slow with this. I've fixed the formatting issue above, and rebased on latest `main`. I think it's good to go, except for the codecov checks